File: TransCoder_tokenized_test_set_functions/python/CHECK_GRAPHS_CYCLE_ODD_LENGTH.py
def containsOdd ( G, src ) :
    global V
    colorArr = [ - 1 ] * V
    colorArr [ src ] = 1
    q = queue . Queue ( )
    q . put ( src )
    while ( not q . empty ( ) ) :
        u = q . get ( )
        if ( G [ u ] [ u ] == 1 ) :
            return True
        for v in range ( V ) :
            if ( G [ u ] [ v ] and colorArr [ v ] == - 1 ) :
                colorArr [ v ] = 1 - colorArr [ u ]
                q . put ( v )
            elif ( G [ u ] [ v ] and colorArr [ v ] == colorArr [ u ] ) :
                return True
    return False

File: results/java_codex/CHECK_GRAPHS_CYCLE_ODD_LENGTH.java
bool containsOdd(int G[][], int src) {
  int colorArr[V];
  for (int i = 0; i < V; ++i) colorArr[i] = -1;
  colorArr[src] = 1;
  queue<int> q;
  q.push(src);
  while (!q.empty()) {
    int u = q.front();
    q.pop();
    if (G[u][u] == 1) return true;
    for (int v = 0; v < V; ++v) {
      if (G[u][v] == 1 && colorArr[v] == -1) {
        colorArr[v] = 1 - colorArr[u];
        q.push(v);
      } else if (G[u][v] == 1 && colorArr[v] == colorArr[u]) return true;
    }
  }
  return false;
}